eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -79,18 +88,37 @@ jobs:
   - checkout: self
     submodules: recursive
 
+  # Install apt dependencies (if appropriate)
+  - bash: sudo apt install -y $(AptDependencies)
+    condition: ne(variables['AptDependencies'], '')
+    displayName: Install dependencies from apt
+
   - template: /eng/pipelines/templates/steps/vcpkg.yml
     parameters:
       DependenciesVariableName: VcpkgInstall
 
+  - script: |
+      dotnet tool install -g dotnet-reportgenerator-globaltool
+      dotnet tool install dotnet-reportgenerator-globaltool --tool-path tools
+    displayName: Install coverage tools
+    condition: ne(variables['CODE_COVERAGE'], '')
+
   - template: /eng/pipelines/templates/steps/cmake-build.yml
     parameters:
-      GenerateArgs: $(CmakeArgs)
+      GenerateArgs: "${{ variables[CmakeArgs] }}"
+      BuildArgs: "${{ variables[BuildArgs] }}"
 
   - script: ctest -C Debug -V --tests-regex ${{ parameters.CtestRegex }}
     workingDirectory: build
     displayName: ctest
 
+  # Make coverage targets (specified in coverage_targets.txt) and assemble
+  # coverage report
+  - template: /eng/pipelines/templates/steps/code-coverage.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+      Enabled: "${{ variables[CODE_COVERAGE] }}"
```

Comment:
This variable is set at runtime so it will always be empty at template expansion time.